app/helpers/ExerciseFileStorage.php

```diff
@@ -31,13 +32,14 @@ public function __construct(FileServerProxy $fileServer) {
   /**
    * Save the file into fileserver and return database entity
    * @param UploadedFile|FileUpload $file The file to be stored
-   * @param Exercise $exercise
-   * @return ExerciseFile|NULL If the operation is not successful, NULL is returned
+   * @param Exercise|null $exercise
+   * @param Pipeline|null $pipeline
+   * @return SupplementaryExerciseFile|NULL If the operation is not successful, NULL is returned
    * @internal param User $user User, who uploaded the file
    */
-  public function store(UploadedFile $file, Exercise $exercise) {
+  public function store(UploadedFile $file, Exercise $exercise = null, Pipeline $pipeline = null) {
```

Comment:
This is pretty ugly - can both arguments be null? It would be better to have two separate methods.